sunpy/map/mapbase.py

```diff
@@ -2305,7 +2348,9 @@ def _process_levels_arg(self, levels):
                                 "it should be an Astropy Quantity object.")
 
         if levels.unit == u.percent:
-            return 0.01 * levels.to_value('percent') * np.nanmax(self.data)
+            # Calculate contour levels based on area containment
+            contour_levels = self._calculate_contour_levels_by_area(levels)
```

Comment:
We can't change the default behavior of this method.

I think for now, adding a "method" keyword (to the draw contours method) that allows you to select this new method would be better.